pkg/portal/graphql/app_mutation.go

```diff
@@ -69,30 +71,35 @@ var _ = registerMutationField(
 
 			gqlCtx := GQLContext(p.Context)
 
-			lazy := gqlCtx.Apps.Get(appID)
-			return lazy.
-				Map(func(value interface{}) (interface{}, error) {
-					app := value.(*model.App)
-					var resourceUpdates []resources.Update
-					for _, f := range updates {
-						f := f.(map[string]interface{})
-						path := f["path"].(string)
-						var data []byte
-						if stringData, ok := f["data"].(string); ok {
-							data = []byte(stringData)
-						}
-
-						resourceUpdates = append(resourceUpdates, resources.Update{
-							Path: path,
-							Data: data,
-						})
-					}
-
-					return map[string]interface{}{
-						"app": gqlCtx.Apps.UpdateResources(app, resourceUpdates),
-					}, nil
-				}).
-				Value, nil
+			app, err := gqlCtx.AppService.Get(appID)
```

Comment:
If I remember correctly we're enforcing access control in loader layer, it seems authz check would be skipped if we call services directly?

Reply:
Maybe we should enforce access control in resolver? It is the place where the context is clear enough.

Reply:
That sounds good to me.

Reply:
Updated.